jujugui/static/gui/src/app/components/entity-details/header/test-header.js

```diff
@@ -124,6 +130,100 @@ describe('EntityHeader', function() {
     assert.deepEqual(output, expected);
   });
 
+  it('renders an old entity properly', function() {
```

Comment:
Thanks for the new test; that said, what we decided awhile back was that we wanted to minimize deepEqual assertions against all the markup as those tests are very fragile and need to be updated with every HTML change. Generally what you'll see is one test (the "it renders" test) that checks all the markup; subsequent tests focus in on specific tidbits. The only time you'll see a test go back and re-assert all the markup is if it's testing a scenario that significantly changes what's being rendered. I'd like to see this test just check to make sure `.entity-header__version`'s text is '#122'.

Now then, this is where things get messy. There are some good libraries out there for querying JSX trees using CSS selectors (Enzyme, http://airbnb.io/enzyme/), unfortunately we don't use them (yet). Consequently we're stuck doing this the hard way:

~~~~
const revision = output.props.children.props.children.props.children[0].props.children[1].props.children[2].props.children[1];
~~~~

As you can see, not nearly as friendly as:

~~~~
const revision = output.find('.entity-header__version').text();
~~~~

But it's what we have at the moment, and it's less fragile/easier to debug (as acknowledged in your specificDeepEquals comments) than full deepEqual assertions against all the markup.